packages/video_player/video_player_avfoundation/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -236,6 +242,90 @@ - (instancetype)initWithPlayerItem:(AVPlayerItem *)item
   return self;
 }
 
+- (void)setPictureInPicture:(BOOL)pictureInPictureEnabled {
+  if (self.pictureInPictureEnabled == pictureInPictureEnabled) {
+    return;
+  }
+
+  self.pictureInPictureEnabled = pictureInPictureEnabled;
+  if (_pipController && self.pictureInPictureEnabled &&
+      ![_pipController isPictureInPictureActive]) {
+    if (_eventSink != nil) {
+      _eventSink(@{@"event" : @"startingPiP"});
+    }
+    [_pipController startPictureInPicture];
+  } else if (_pipController && !self.pictureInPictureEnabled &&
+             [_pipController isPictureInPictureActive]) {
+    [_pipController stopPictureInPicture];
+  }
+}
+
+#if TARGET_OS_IOS
+- (void)setupPipController:(BOOL)canStartPictureInPictureAutomaticallyFromInline {
+  if ([AVPictureInPictureController isPictureInPictureSupported]) {
+    _pipController = [[AVPictureInPictureController alloc] initWithPlayerLayer:self.playerLayer];
+    if (@available(iOS 14.2, *)) {
+      _pipController.canStartPictureInPictureAutomaticallyFromInline =
+          canStartPictureInPictureAutomaticallyFromInline;
+    }
+    _pipController.delegate = self;
+  }
+}
+
+- (void)usePlayerLayer:(CGRect)frame
+    canStartPictureInPictureAutomaticallyFromInline:
+        (BOOL)canStartPictureInPictureAutomaticallyFromInline {
+  if (_player) {
+    self.playerLayer = [AVPlayerLayer playerLayerWithPlayer:_player];
+    UIViewController *vc = [[[UIApplication sharedApplication] keyWindow] rootViewController];
```

Comment:
i believe this is deprecated due to iPad?

Reply:
^ This too. `keyWindow` is already deprecated

Reply:
I saw that you already added the new implementation on iOS 16 so I changed my code to use your implementation. I still use this for old versions.